### ISSUE TITLE
SecItemCopyMatching not works in iphone 12

### ISSUE THREAD

**Issue by MajidAlNuumani** [agens-no/EllipticCurveKeyPair#53]
I face issue with Face Id in **_some_** of  iphone 12 pro and pro max the issue happen in SecKeyCreateSignature not generate signature but it works fine in different iphones like iphone 11.

even I cannot get the Face Id prompt.

my app not show in Face id & Passcode -> other apps.

In iphone 12 pro not in all device happen only a few devices.

**Comment by hfossli** [comment]
Threading issues?

Håvard

**Comment by MajidAlNuumani** [comment]
I am not sure but here is my code:

Singing data code:
`
    private func Signing(dataToBeSigned: String) -> String {
        do {
            let digest = dataToBeSigned.data(using: .utf8)!
            let singedData = try Shared.keypair.sign(digest, hash: .sha256)
            let signature = singedData.base64EncodedString()
            return signature
        } catch {
            return "Cannot sign data"
        }
    }`

I call it from this code:
`if(call.method == "Register") {
            guard let args = call.arguments else {
              return
            }
            if let myArgs = args as? [String: Any],
               let blaBla1 = myArgs["blaBla1"] as? String,
               let BlaBla2 = myArgs["BlaBla2"] as? String {
               let signatureBla1 = self?.Signing(dataToBeSigned: blaBla1)
               let signatureBla2 = self?.Signing(dataToBeSigned: BlaBla2)
                let publicKey = self?.getPublicKey()
                var list = [String]()
                list.append(publicKey!)
                list.append(signatureBla1!)
                list.append(signatureBla2!)
              result(list)
            } else {
              result("iOS could not extract flutter arguments in method: (sendParams)")
            }
            self?.Register(result: result)`

`call.method == "Register"` is use to call Register Function from Flutter using MethodChannel.

Can you help if this issue happen because of threads?

**Comment by hfossli** [comment]
I would check that all the if let's are valid and unwrapped correctly. I don't understand if this compiles because dataToBeSigned expects Data not String

**Comment by MajidAlNuumani** [comment]
Yes but I convert it to Data in Signing function here `dataToBeSigned.data(using: .utf8)!`

**Comment by MajidAlNuumani** [comment]
Hello All,

I got an update in this thread.

I got status error -25300 when call this:

`let status = SecItemCopyMatching(query as CFDictionary, &raw)`

Any help why this happen in some of iPhone 12 pro and most of iPhone 12 pro max?

**Comment by MajidAlNuumani** [comment]
Can any one help in this issue?
